internal/commands/root/http.go

```diff
@@ -125,6 +128,12 @@ func setupHTTPServer(ctx context.Context, p provider.Provider, cfg *apiServerCon
 		if mp, ok := p.(provider.PodMetricsProvider); ok {
 			podRoutes.GetStatsSummary = mp.GetStatsSummary
 		}
+
+		if cfg.AuthWebhookEnabled && cfg.Auth != nil {
+			m := NewVirtualKubeletAuthFilter(ctx, cfg.Auth)
+			podRoutes.AuthHandler = m.AuthHandler
```

Comment:
Here, instead of setting an `AuthHandler we can replace `mux` with a wrapped implementation that injects the auth handler.

Since `mux` just needs to implement this:

```go
type ServeMux interface {
	Handle(path string, h http.Handler)
}
```

Reply:
@cpuguy83 change is done, please review again.

Reply:
I will abandon the pr on virtual-kubelet/virtual-kubelet, it is not needed anymore.